### ISSUE TITLE
`one-var-declaration-per-line`: incorrect fix applied when combined with `one-var`

### ISSUE THREAD

**Issue by u01jmg3** [eslint-stylistic/eslint-stylistic#593]
### Validations

- [X] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.

### Describe the bug

_Previously logged and closed here: https://github.com/eslint/eslint/issues/19081_

---

### Environment

Node version: v22.11.0
npm version: v10.9.0
Local ESLint version: N/A
Global ESLint version: v9.13.0
ESLint Stylistic version: v2.10.0
Operating System: Win 11 x64

### What did you do?

```js
/*rules: {
    'one-var': [
        'error',
        'never',
    ],
    '@stylistic/one-var-declaration-per-line': [
        'error',
        'always',
    ],
}*/

var buttonsHTML = '', buttonClass, button;
```

### What did you expect to happen?

```js
var buttonsHTML = ''; 
var buttonClass;
var button;
```

### What actually happened?

```js
var buttonsHTML = ''; var buttonClass; var
    button;
```

### Additional comments

The issue has to do with the order the 3 fixes are applied. If you apply them manually on ESLint playground you can get the desired output.

### Reproduction

https://eslint.org/play/#eyJ0ZXh0IjoiLyplc2xpbnQgb25lLXZhcjogW1wiZXJyb3JcIiwgXCJuZXZlclwiXSovXG4vKmVzbGludCBvbmUtdmFyLWRlY2xhcmF0aW9uLXBlci1saW5lOiBbXCJlcnJvclwiLCBcImFsd2F5c1wiXSovXG5cbnZhciBidXR0b25zSFRNTCA9ICcnLCBidXR0b25DbGFzcywgYnV0dG9uOyIsIm9wdGlvbnMiOnsicnVsZXMiOnt9LCJsYW5ndWFnZU9wdGlvbnMiOnsic291cmNlVHlwZSI6Im1vZHVsZSIsInBhcnNlck9wdGlvbnMiOnsiZWNtYUZlYXR1cmVzIjp7fX19fX0=

### Contributes

- [ ] I am willing to submit a PR to fix this issue
- [ ] I am willing to submit a PR with failing tests
